app/models/service/aggregation.rb

```diff
@@ -12,15 +12,15 @@ module Service::Aggregation
     virtual_attribute :aggregate_all_vm_cpus,   :integer,
                       :arel => aggregate_service_hardware_arel('cpus', :cpu_total_cores)
     virtual_attribute :aggregate_all_vm_memory, :integer,
-                      :arel => aggregate_service_hardware_arel('memory', :memory_mb)
+                      :arel => aggregate_service_hardware_arel('memory', :ram_size_in_bytes)
```

Comment:
this one I don't know.

@NickLaMuro would have better insight if it works with virtual attributes

I'm tempted to say that we should use memory_mb and then modify nick's code to use `Model.attribute_arel(column)` instead of `[column]`

Reply:
he's gonna _love_ getting dragged into this one

Reply:
> ~~he's gonna love getting dragged into this one~~ I mean, sure, that sounds good

You were not wrong...

Reply:
> this one I don't know.

Not sure of the context here of this and what Keenan's concern is that requires my insight.

> would have better insight if it works with virtual attributes

That was the intent I am pretty sure, otherwise I wouldn't have bothered with all of the Arel.  Though, maybe I am a bit confused on what "works with virtual attributes" is supposed to mean.

From what I do remember off the top of my head, one of the big "gotchas" of this `module` is that there was a limit in the database for the size of integers that was less than what ruby supports.  If we are working with bytes, we probably would exceed that limit pretty quick, so if you are doing some `SUM` functions in a DB call, this might have issues.

But I am still playing catchup on the PR and what is being attempted here.

Reply:
Sorry to bother you @NickLaMuro 
@d-m-u will add tests and see if this is broken

Reply:
> Sorry to bother you @NickLaMuro

Don't worry, I am mostly being sarcastic and difficult.  You are probably right to get me involved at least a bit.

That said... didn't really answer the questions I posed above...